### PR TITLE
prometheus plugin

### PR DESCRIPTION
**Pull request by nataly87s**

### CODE REVIEW

File: prometheus/index.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
+const program = require('commander');
+const axios = require('axios');
+const querystring = require('querystring');
+const Plugin = require('nagios-plugin');
+const package = require('./package.json');
+
+let endpoint, query;
+
+const funnelPlugin = new Plugin({
```
Comment (yaron-idan) [inline prometheus/index.js]: It's not a funnelPlugin, change to appropriate name

**Comment by omerlh** [comment]
Awesome work, thanks! @yaron-idan @oryagel (and Nataly, of course) I think that:
- We should move toward repository per plugin
- Maybe we should supply docker file and docker build for each plugin? We can start with this plugin, of course.

What do you think?

File: prometheus/index.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  shortName: 'prometheus'
+});
+
+program
+  .version(package.version)
+  .option('-e, --endpoint <prometheus_endpoint>', 'Prometheus endpoint (required)')
+  .option('-w, --warning <warning>', 'Warning threshold', 0)
+  .option('-c, --critical <critical>', 'Critical threshold', 0)
+  .option('-t, --time <rfc3339 | unix_timestamp>', 'Evaluation timestamp (optional)')
+  .arguments('<query>')
```
Comment (yaron-idan) [inline prometheus/index.js]: Why not add the method arg like the original plugin?

File: prometheus/index.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+function handleResponse(response) {
+  if (response.data.status !== 'success') {
+    console.error('Prometheus responded with an error', JSON.stringify(response.data));
+    process.exit(funnelPlugin.states.UNKNOWN);
+  }
+
+  const data = response.data.data;
+  const { resultType, result: [_, result] } = response.data.data
+
+  if (resultType !== 'scalar') {
```
Comment (yaron-idan) [inline prometheus/index.js]: Maybe we should reverse the condition to make the plugin extensible for when we want to support vectors.

File: prometheus/index.js (hunk 2 above)
Comment (nataly87s) [reply prometheus/index.js]: I just liked this syntax because it's required

Comment (nataly87s) [reply prometheus/index.js]: because we can know that from the warning/critical parameters

**Pull request merged**